### ISSUE TITLE
What happened to this page?

### ISSUE THREAD

**Issue by Air-Git** [MicrosoftDocs/OfficeDocs-SkypeForBusiness#5368]
The page does not cover what it says in the title, and has not been updated since 2017. There seems to be no documentation on how to configure Exchange on-premises so that Teams can access a user's calendar.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 20c0a619-c505-73a3-d815-58b384dc9a61
* Version Independent ID: 78b91880-6d1e-a9dc-6099-7a0a15b96a4a
* Content: [Configure an Exchange hybrid organization - Microsoft Teams](https://docs.microsoft.com/en-us/microsoftteams/exchange-hybrid-organization)
* Content Source: [Teams/Exchange-hybrid-organization.md](https://github.com/MicrosoftDocs/OfficeDocs-SkypeForBusiness/blob/live/Teams/Exchange-hybrid-organization.md)
* Service: **msteams**
* GitHub Login: @dstrome
* Microsoft Alias: **dstrome**

**Comment by scanum** [comment]
@Air-Git  Thank you for submitting feedback and contributing to the docs. We are currently investigating this.

**Comment by dariomws** [comment]
Hi @Air-Git, thank you for your feedback and help us to improve docs.microsoft.com.

For information about how to configure Exchange on-premises so that Teams can access a user's calendar, please check the following article:
https://docs.microsoft.com/en-us/microsoftteams/exchange-teams-interact

If you're having trouble getting it to work, check out the following article:
https://techcommunity.microsoft.com/t5/exchange-team-blog/configuring-teams-calendar-access-for-exchange-on-premises/ba-p/1484009

**Comment by Air-Git** [comment]
Thanks. The [How Exchange and Teams Interact](https://docs.microsoft.com/en-us/microsoftteams/exchange-teams-interact) document does not cover what's required. But the Exchange team blog published on 23 June covers the gaps. That's the only document I've found that describes accurately what you need to do to integrate the Teams calendar with Exchange on-premises.

**Comment by dariomws** [comment]
@Air-Git, in summary, you need:
- Exchange Hybrid (AAD Connect + HCW)
- Mailbox located on Exchange 2016 CU3 or later

There are features that require additional steps:
- Calendar delegation
- M365 groups sync

**Comment by Air-Git** [comment]
I don't think this is strictly true. You need OAuth authentication between the Azure AD service principles and the on-premises Exchange and Skype for Business services. And you need inbound access to the Autodiscover and EWS services. Using the HCW is one way to get OAuth, but in fact for a long time the HCW failed to do this and you had to do it manually. If you do it via the HCW, you also get various services like the migration proxy, which obviously you don't need to get access to the calendar. Also, the HCW method only works if you choose the full classic hybrid. Teams cannot access the Exchange on-premises calendar if you use the Modern Hybrid approach, with the Hybrid agent, because this has no inbound access to the mailbox servers.

**Comment by dariomws** [comment]
Hi @Air-Git,

It´s recommended that you configure OAuth using the HCW as the other article states:
"providing Exchange Server 2016 (Cumulative Update 3), or later, is running on-premises with OAuth configured (preferably via the Exchange Hybrid Configuration Wizard) as described in Configure OAuth authentication between Exchange and Exchange Online organizations"

Autodiscover and EWS services are a requirement in hybrid environments, for example, to have free/busy working or multi-search: https://docs.microsoft.com/exchange/hybrid-deployment-prerequisites

Group writeback is not a requirement for Teams access to the Exchange calendar.

In summary, to configure Teams calendar access for exchange on-premises "following best practices" you need:
- Exchange Hybrid (https://docs.microsoft.com/exchange/hybrid-deployment-prerequisites, https://docs.microsoft.com/exchange/exchange-hybrid)
- Mailbox located on Exchange 2016 CU3 or later

**Comment by Air-Git** [comment]
This reference to needing a hybrid is misleading and not accurate. There are five types of hybrid. Only one of the five will enable Teams calendar access to Exchange 2016 on-premises mailboxes.

The solution is that you need the features associated with a [Full Classic Hybrid](https://docs.microsoft.com/en-us/exchange/hybrid-configuration-wizard-options). This is not because you need the hybrid, but because that particular hybrid model uses the same features:

1. OAuth authentication relationship
2. AAD Connect Exchange hybrid option
3. Public access to EWS and Autodiscover
4. Mailboxes on Exchange Server 2016.

The Exchange Team Blog that you referenced is the only Microsoft document I've found that describes this accurately.

**Comment by dariomws** [comment]
@Air-Git, yes, to configure OAuth you need to select a full hybrid option (Classic or Modern). We recommend that you configure it using the HCW, but configuring it manually is supported sure.

**Comment by Air-Git** [comment]
But Modern Full does not have the inbound access to Autodiscover and EWS. That's why I think the reference to Hybrid is a bit of a red herring. Only the Classic Full hybrid has those features. And you don't even need the hybrid. You just need similar configurations.

The Teams document hints at this, when it says that only the Classic Full Hybrid will provide access to the Teams calendar. Unfortunately, although it says you need EWS and Autodiscover, it does not explicitly say you need external/public/internet access to them. Many people have EWS and Autodiscover, but internally and not externally.

**Comment by dariomws** [comment]
Modern full hybrid doesn't require autodiscover and EWS. You can have them or not. (Yes,  they are one of the reasons to use modern hybrid, avoid expose them to Internet)
